src/main/java/com/thoughtworks/gauge/scan/StepsScanner.java

```diff
@@ -51,11 +51,26 @@ private void buildStepRegistry(Set<Method> stepImplementations) {
                 for (String stepName : annotation.value()) {
                     String parameterizedStep = Util.trimQuotes(stepName);
                     String stepText = stepsUtil.getStepText(parameterizedStep);
-                    List<String> parameters = stepsUtil.getParameters(parameterizedStep);
-                    StepValue stepValue = new StepValue(stepText, parameterizedStep, parameters);
-                    registry.addStepImplementation(stepValue, method);
+                    if (registry.contains(stepText)) {
+                        StepRegistryEntry entry = registry.get(stepText, method);
+                        if (entry != null) {
+                            Logger.debug("Found " + stepText + " in current project scope.");
+                            entry.setMethodInfo(method);
+                        } else {
+                            addStepEntryToRegistry(stepsUtil, method, parameterizedStep, stepText);
+                        }
+                    } else {
+                        addStepEntryToRegistry(stepsUtil, method, parameterizedStep, stepText);
+                    }
```

Comment:
Is it possible to write a unit test for this section?

Reply:
added tests